File: go/modcdp/launcher/NoopBrowserLauncher.go
package launcher

type NoopBrowserLauncher struct {
	BrowserLauncher
}

func NewNoopBrowserLauncher(options LaunchOptions) *NoopBrowserLauncher {
	return &NoopBrowserLauncher{BrowserLauncher: NewBrowserLauncher(options)}
}

func (l *NoopBrowserLauncher) Launch(options LaunchOptions) (*LaunchedBrowser, error) {
	launched := &LaunchedBrowser{Close: func() {}}
	l.Launched = launched
	return launched, nil
}

File: python/modcdp/launcher/NoopBrowserLauncher.py
from __future__ import annotations

from ..launcher.BrowserLauncher import BrowserLaunchOptions, BrowserLauncher, LaunchedBrowser


class NoopBrowserLauncher(BrowserLauncher):
    def launch(self, options: BrowserLaunchOptions | None = None) -> LaunchedBrowser:
        self.launched = {"cdp_url": None, "close": lambda: None}
        return self.launched
